jerry-core/jerryscript.h

```diff
@@ -389,6 +389,11 @@ size_t jerry_parse_and_save_literals (const jerry_char_t *source_p, size_t sourc
                                       uint32_t *buffer_p, size_t buffer_size, bool is_c_format);
 
 /**
+ * External context functions.
+ */
+size_t jerry_get_context_size (void);
+size_t jerry_get_lcache_size (size_t row_count, size_t row_length);
```

Comment:
I think that's a consequence of going for 3 separate allocations for heap, lcache and context.
See discussion here https://github.com/jerryscript-project/jerryscript/pull/1778/files#r113155225
(Somehow I interpreted your answers there as being in favor of the making the user allocate 3 separate memory blocks.)

Suggestions on how to simplify the API?

I had suggested something like this [before]:

To create a context, we could also have an API like:
```
typedef struct {
  // user's allocator function, will be called multiple times during jerry_instance_create().
  // semantics are the same as malloc.
  void *(*alloc)(size_t size_in_bytes);

  // JS heap size. Leave zero to pick default size.
  size_t heap_size;

  // LCache sizes. Leave any zero to pick the default size.
  struct {
   size_t row_count;
   size_t row_length;
  } lcache;
} jerry_instance_config_t;

jerry_instance_t *jerry_instance_create(const jerry_instance_config_t *config);
```
`jerry_instance_t *` can be a fwd declared struct.
This way we're spilling less of the internals.

@zherczeg what do you think about that?

[before]: https://github.com/jerryscript-project/jerryscript/issues/1746#issuecomment-295645286